File: src/c-qsort.c
int jsArray[] = {5, 1, 3, 2, 4, 0};

void myqsort(int left, int right)
{
  if(left > right){
    return;
  }
  int tmp = jsArray[left];
  int i = left;
  int j = right;
  while(i != j){
    while(jsArray[j] >= tmp && j > i) j--;
    while(jsArray[i] <= tmp && j > i) i++;
    int t = jsArray[i];
    jsArray[i] = jsArray[j];
    jsArray[j] = t;
  }
  jsArray[left] = jsArray[i];
  jsArray[i] = tmp;

  myqsort(left, i-1);
  myqsort(i+1, right);
}

int getter(int index){
  return jsArray[index];
}

File: src/c-qsort.ts
import fs from "node:fs";

async function getResult() {
    const mydir = __dirname + "/../build/myc-qsort.wasm";
    const content = await fs.promises.readFile(mydir);
    
    const module = await WebAssembly.instantiate(content);
    const myqsort = module.instance.exports.myqsort as CallableFunction;
    const getter = module.instance.exports.getter as CallableFunction;
    // const memm = module.instance.exports.memory as WebAssembly.Memory;
    // console.log(memm.buffer);
    // const a =  memm.buffer.slice(1024);
    // console.log(a);

    const rArray = new Array<number>();
    myqsort(0, 5);

    for (let index = 0; index < 6; index++){
        rArray[index] = getter(index);
    }

    return rArray;
}
export const processor = {getResult};
